File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        int n = nums.size();

        vector<pair<int,int>>temp;

        for(int i = 0; i < n; i++){
            temp.push_back({nums[i] , i});
        }

        sort(temp.begin() , temp.end());

        int left = 0 , right = n - 1;

        while(left < right){

            int a = temp[left].first;
            int b = temp[right].first;

            if((a + b) == target){
                return {temp[left].second , temp[right].second};
            }

            else if((a + b) < target){
                left++;
            }

            else{
                right--;
            }
        }

        return {-1};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        n = len(nums)

        d = dict()

        for i in range(n):
            remaining = target - nums[i]
            if(d.get(remaining , None) != None):
                return [i , d[remaining]]

            d[nums[i]] = i
        
        return [-1 , -1]